### PR TITLE
Change remaining misuse of Ticket terminology.

### PR DESCRIPTION
**Pull request by MaggieYingYi**
After discussing the issue with the team, we decided to do the following changes:

RepoTicket.cpp =>RepoDefinition.cpp
RepoTicket.h => RepoDefinition.h
ticketmd => repodefinition
TicketNode => RepoDefinition

Fix for <https://github.com/SNSystems/llvm-project-prepo/issues/8>.

### CODE REVIEW

**Comment by paulhuggett** [comment]
> After discussing the issue with the team

It would be great to record that discussion in a github issue.

**Comment by MaggieYingYi** [comment]
> 
> 
> > After discussing the issue with the team
> 
> It would be great to record that discussion in a github issue.

Good point. I have put the discussion in the https://github.com/SNSystems/llvm-project-prepo/issues/8#issuecomment-622106506.

File: llvm/test/Assembler/invalid-ticketnode-linkage.ll [hunk 1]
```diff
@@ -1,5 +0,0 @@
-; RUN: not llvm-as < %s 2>&1 | FileCheck %s
-
-@v = global i32 1, align 4, !fragment !0
-!0 = !TicketNode(name: "v", digest: [16 x i8] c"\DC\8BWeQ\E4\03\E6\F3:\DE\D1\9F\90\AC\F7", linkage: internal_global)
-; CHECK: Invalid linkage type!
```
Comment (paulhuggett) [inline llvm/test/Assembler/invalid-ticketnode-linkage.ll:5]: Why was this file deleted?

Comment (paulhuggett) [reply llvm/test/Assembler/invalid-ticketnode-linkage.ll:5]: Scratch that. I now see that it was renamed. Git/Github wasn't able to spot that.

File: llvm/include/llvm/IR/RepoDefinition.h [hunk 2]
```diff
@@ -28,27 +28,24 @@ class dependents;
 
 namespace llvm {
 
-struct Ticket;
-
-/// Global value ticket metadata description.
-namespace ticketmd {
+/// Global object repo definition description.
```
[the remark this answers is not in the document]
Comment (MaggieYingYi) [reply llvm/include/llvm/IR/RepoDefinition.h]: Fixed, thanks.

File: llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp [hunk 3]
```diff
@@ -260,20 +260,21 @@ bool AsmPrinter::doInitialization(Module &M) {
   const Triple &TT = TM.getTargetTriple();
   // Pass global metadata 'repo.metadata' to OutContext.
   if (TT.isOSBinFormatRepo()) {
-    if (NamedMDNode *Globals = M.getNamedMetadata("repo.tickets")) {
+    if (NamedMDNode *Globals = M.getNamedMetadata("repo.definitions")) {
       for (auto MDN : Globals->operands()) {
-        if (TicketNode *TN = dyn_cast<TicketNode>(MDN)) {
-          OutContext.addTicketNode(TN);
+        if (RepoDefinition *RD = dyn_cast<RepoDefinition>(MDN)) {
```
[the remark this answers is not in the document]
Comment (MaggieYingYi) [reply llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp]: Fixed, thanks.

File: llvm/include/llvm/IR/RepoDefinition.h [hunk 4]
```diff
@@ -1,4 +1,4 @@
-//===- RepoTicket.h - Program repository digest data structure. -*- C++ --===//
+//===- RepoDefinition.h - Program repository metadata definitions. *-C++-*-===//
```
[the remark this answers is not in the document]
Comment (MaggieYingYi) [reply llvm/include/llvm/IR/RepoDefinition.h]: Fixed, thanks.

File: llvm/test/Assembler/invalid-ticketnode-linkage.ll (hunk 1 above)
Comment (MaggieYingYi) [reply llvm/test/Assembler/invalid-ticketnode-linkage.ll:5]: Ok, I see :-)
